src/schemas/api-reference.json

```diff
@@ -161,7 +166,12 @@
                     },
                     "scopes": {
                         "description": "Scope expression template specifying required scopes for a method. Not provided if authentication isn't required.",
-                        "$ref": "#/definitions/scopeExpressionTemplate"
+                        "oneOf": [
```

Comment:
Why is this not just scopeExpressionTemplate?  We allow "if" at the top level but not "for"?

Reply:
I think `for` is not allowed at top level because it evaluates to an array of strings rather than a scope expression itself.

Reply:
I think jonas is open to changing that however.

Reply:
And maybe the more I think about it I like it more.

Reply:
No, that reason makes sense.  I'd say "add a comment" but of course, it's JSON :)

Reply:
So there is another thing we could do, instead of for and if, we make them conditions on AllOf/AnyOf, like:

`{AllOf: [...], if: 'param'}` and 
`{AllOf: '...<var>...', for: 'var', in: 'param'}`

This way we don't have to merge upwards when evaluating... and AnyOf/AllOf are used to indicate if the condition is trivial satisfied or rejected if the array in "for" is empty or the condition in "if" false...

It would make if/for side conditions on the AllOf/AnyOf constructs.... I'm not sure it's more logical. Which is what matters as it's purely a syntactical thing...

Which is also why I stopped thinking about it... As we could also start questioning why AllOf and AnyOf starts with a capital letter...

Reply:
They start with a capital letter only because they did in the spec in rfc-23. I don't mind either way.